ch2.2.3-ex7/reverse-linked-list.bel

```diff
@@ -0,0 +1,19 @@
+(tem linked-list contents nil)
+
+(def prs-linked-list (lst)
+  (prs "<linked-list " lst!contents ">"))
+
+(def rev-linked-list (lst)
+  (let last (rev-linked-node lst!contents)
+    (set lst!contents last)))
+
+(def rev-linked-node (node (o prev))
+  (if node
```

Comment:
As much as `if node` makes the code one line shorter, I think it's more idiomatic to reverse the test and write it as `if (no node)`, with `nil` as the consequent.

The way it is not, `nil` is still the consequent, but that's implicit in the semantics of `if`, not explicit in the code. Hiding the base case like that may score points for brevity, but not for clarity. Besides, I count 43 hits on the search `if (no` in bel.bel, but I can't recall seeing the corresponding `if` form anywhere.